src/traced/probes/sys_stats/sys_stats_data_source.cc

```diff
@@ -406,25 +420,47 @@ std::optional<uint64_t> SysStatsDataSource::ReadAMDGpuFreq() {
 }
 
 void SysStatsDataSource::ReadGpuFrequency(protos::pbzero::SysStats* sys_stats) {
-  // For Adreno GPUs.
-  auto freq = ReadFileToUInt64("/sys/class/kgsl/kgsl-3d0/devfreq/cur_freq");
-  if (freq) {
+  if (adreno_gpufreq_fd_) {
+    auto freq = ReadFileToUInt64(&adreno_gpufreq_fd_,
+                                 "/sys/class/kgsl/kgsl-3d0/devfreq/cur_freq");
     sys_stats->add_gpufreq_mhz((*freq) / 1'000'000);
-    return;
   }
-
-  // For Intel GPUs.
-  freq = ReadFileToUInt64("/sys/class/drm/card0/gt_act_freq_mhz");
+  for (auto& [fd, path] : intel_gpufreq_fds_) {
+    auto freq = ReadFileToUInt64(&fd, path.c_str());
+    if (freq) {
+      sys_stats->add_gpufreq_mhz((*freq));
+    }
+  }
+  auto freq = ReadAmdGpuFreq();
   if (freq) {
     sys_stats->add_gpufreq_mhz(*freq);
-    return;
   }
+}
 
-  // For AMD GPUs.
-  freq = ReadAMDGpuFreq();
-  if (freq) {
-    sys_stats->add_gpufreq_mhz(*freq);
+void SysStatsDataSource::OpenGpuFreqFiles(OpenFunction open_fn) {
+  for (size_t tile = 0;; ++tile) {
+    bool found = false;
+    for (size_t gt = 0;; ++gt) {
```

Comment:
maybe pick a different var name? the chances of some system header doing `#define gt ` are non 0 given all the random platforms we support these days (I have scars from people including <iso646.h>`  in another life)